src/FSharp.Control.TaskSeq.Test/TaskSeq.SkipWhile.Tests.fs

```diff
@@ -0,0 +1,315 @@
+            |> TaskSeq.toArrayAsync
+
+        let expected = [| 5..10 |]
+        first |> should equal expected
+
+        // side effect, reiterating causes it to resume from where we left it (minus the failing item)
+        // which means the original sequence has now changed due to the side effect
+        let! repeat =
+            TaskSeq.skipWhile (fun x -> x < 5) ts
+            |> TaskSeq.toArrayAsync
+
+        repeat |> should not' (equal expected)
+    }
+
+    [<Theory; ClassData(typeof<TestSideEffectTaskSeq>)>]
+    let ``TaskSeq-skipWhileInclusiveAsync consumes the prefix for a longer sequence, with mutation`` variant = task {
+        let ts = Gen.getSeqWithSideEffect variant
+
+        let! first =
+            TaskSeq.skipWhileInclusiveAsync (fun x -> task { return x < 5 }) ts
+            |> TaskSeq.toArrayAsync
+
+        let expected = [| 6..10 |]
+        first |> should equal expected
+
+        // side effect, reiterating causes it to resume from where we left it (minus the failing item)
+        // which means the original sequence has now changed due to the side effect
+        let! repeat =
+            TaskSeq.skipWhileInclusiveAsync (fun x -> task { return x < 5 }) ts
+            |> TaskSeq.toArrayAsync
+
+        repeat |> should not' (equal expected)
+    }
+
+module Other =
+    [<Fact>]
+    let ``TaskSeq-skipWhileXXX should include all items after predicate fails`` () = task {
+        do!
+            [ 1; 2; 2; 3; 3; 2; 1 ]
+            |> TaskSeq.ofSeq
+            |> TaskSeq.skipWhile (fun x -> x <= 2)
+            |> verifyDigitsAsString "CCBA"
+
+        do!
+            [ 1; 2; 2; 3; 3; 2; 1 ]
+            |> TaskSeq.ofSeq
+            |> TaskSeq.skipWhileInclusive (fun x -> x <= 2)
+            |> verifyDigitsAsString "CBA"
+
+        do!
+            [ 1; 2; 2; 3; 3; 2; 1 ]
+            |> TaskSeq.ofSeq
+            |> TaskSeq.skipWhileAsync (fun x -> Task.fromResult (x <= 2))
+            |> verifyDigitsAsString "CCBA"
+
+        do!
+            [ 1; 2; 2; 3; 3; 2; 1 ]
+            |> TaskSeq.ofSeq
+            |> TaskSeq.skipWhileInclusiveAsync (fun x -> Task.fromResult (x <= 2))
+            |> verifyDigitsAsString "CBA"
+    }
+
+    [<Fact>]
+    let ``TaskSeq-skipWhileXXX stops consuming after predicate fails`` () =
+        let testSkipper skipper =
+            fun () ->
+                seq {
+                    yield! [ 1; 2; 2; 3; 3 ]
+                    yield SideEffectPastEnd "Too far" |> raise
+                }
+                |> TaskSeq.ofSeq
+                |> skipper
+                |> consumeTaskSeq
+            |> should throwAsyncExact typeof<SideEffectPastEnd>
+
+        testSkipper (TaskSeq.skipWhile (fun x -> x <= 2))
+        testSkipper (TaskSeq.skipWhileInclusive (fun x -> x <= 2))
+        testSkipper (TaskSeq.skipWhileAsync (fun x -> Task.fromResult (x <= 2)))
+        testSkipper (TaskSeq.skipWhileInclusiveAsync (fun x -> Task.fromResult (x <= 2)))
```

Comment:
Is there coverage (maybe tehre already is) re predicate flipping on the last one and/or second last item ?

Reply:
Yes, we have these tests.